examples/named_cluster/kms.tf

```diff
@@ -0,0 +1,29 @@
+resource "azurerm_key_vault_key" "kms" {
+  key_opts = [
+    "decrypt",
+    "encrypt",
+    "sign",
+    "unwrapKey",
+    "verify",
+    "wrapKey",
+  ]
+  key_type        = "RSA"
+  key_vault_id    = azurerm_key_vault.des_vault.id
+  name            = "etcd-encryption"
+  expiration_date = timeadd("${formatdate("YYYY-MM-DD", timestamp())}T00:00:00Z", "168h")
+  key_size        = 2048
+
+  depends_on = [
+    azurerm_key_vault_access_policy.current_user
+  ]
+}
+
+resource "azurerm_key_vault_access_policy" "kms" {
+  key_vault_id = azurerm_key_vault.des_vault.id
+  object_id    = azurerm_user_assigned_identity.test.id
```

Comment:
We've got an e2e test error:

```text
TestExamplesNamedCluster 2023-01-27T04:53:34Z logger.go:66: │ Error: expected "object_id" to be a valid UUID, got /subscriptions/f7a632a5-49db-4c5e-9828-cd62cb753971/resourceGroups/8df3563089d5cbb3-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/8df3563089d5cbb3-identity
TestExamplesNamedCluster 2023-01-27T04:53:34Z logger.go:66: │ 
TestExamplesNamedCluster 2023-01-27T04:53:34Z logger.go:66: │   with azurerm_key_vault_access_policy.kms,
TestExamplesNamedCluster 2023-01-27T04:53:34Z logger.go:66: │   on kms.tf line 23, in resource "azurerm_key_vault_access_policy" "kms":
TestExamplesNamedCluster 2023-01-27T04:53:34Z logger.go:66: │   23:   object_id    = azurerm_user_assigned_identity.test.id
```

Could you please try `object_id = azurerm_user_assigned_identity.test.principal_id`, then try to pass `TestExamplesNamedCluster` test inside `test/e2e` folder? Thanks!

Reply:
Cannot run E2E here as I have no access to a unrestricted Azure Subscription.

But jup, `principal_id` will fix it halfway, we also need a `depends_on` inside the call of the module to ensure the managed identity has the correct access policy applied before we call the managedCluster API.

Reply:
done